### PR TITLE
ENH: Add data fetcher

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add data fetcher. Add the accompanying test.

Add a script to fetch a given dataset to the provided local path. Add
the accompanying test. Add the script to the project scripts in the
`pyproject.toml` file.

Add the badge corresponding to the Zenodo record DOI to the `README`.

Explain in the `README` file how to:
- Fetch the data automatically.
- Cite the data.

Enable the package testing and coverage report steps in the
`GitHub Actions` workflow file.

### CODE REVIEW

File: tractolearn/tractoio/dataset_fetch.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+import enum
+import os
+import subprocess
+import sys
+import tarfile
+import zipfile
+from hashlib import md5
+from os.path import join as pjoin
+from shutil import copyfileobj
+from urllib.request import urlopen
```
Comment (jhlegarreta) [inline tractolearn/tractoio/dataset_fetch.py:12]: Requirements to be added.

File: tractolearn/tractoio/dataset_fetch.py [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+
+
+fetch_contrastive_ae_weights = (
+    "fetch_contrastive_ae_weights",
+    TRACTOLEARN_DATASETS_URL + "2xmgw/",
+    ["download"],
+    ["weights.pt"],
+    ["7170d0192fa00b5ef069f8e7c274950c"],
+    "Download contrastive-loss trained autoencoder weights",
+    "543B",
```
Comment (jhlegarreta) [inline tractolearn/tractoio/dataset_fetch.py]: The data needs to be adjusted once it is uploaded to OSF.

File: tractolearn/tractoio/dataset_fetch.py [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+    path : string
+        Destination path.
+
+    Returns
+    -------
+    fnames : string or list
+        Filenames for dataset.
+    """
+
+    print(f"\nDataset: {name}")
```
Comment (jhlegarreta) [inline tractolearn/tractoio/dataset_fetch.py]: Change to logging once a logger is in place,

**Comment by jhlegarreta** [comment]
Eventually, I would like to switch to a versioning-based system. OSF is not very practical for that, but it does a sufficient job as a data hosting service for now.

**Comment by jhlegarreta** [comment]
@felixdumais This should work as is for the test data in zenodo (maybe some requirements need to be added). You can check the test locally if you want. Once PR #33 merged and the testing uncommented, the test should demonstrate that it works. Maybe the 74 GB data will pose a challenge for the download (e.g. a timeout).

File: tractolearn/tractoio/dataset_fetch.py (hunk 2 above)
Comment (jhlegarreta) [reply tractolearn/tractoio/dataset_fetch.py]: Done in [4bdf06b](https://github.com/scil-vital/tractolearn/commit/4bdf06b6316e08a540e14453ab68cc4136866f5c).

File: tractolearn/tractoio/dataset_fetch.py (hunk 1 above)
Comment (jhlegarreta) [reply tractolearn/tractoio/dataset_fetch.py:12]: They are part of the standard Python library, so no need.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # tractolearn
 
 [![test, package](https://github.com/scil-vital/tractolearn/actions/workflows/test_package.yml/badge.svg?branch=main)](https://github.com/scil-vital/tractolearn/actions/workflows/test_package.yml?query=branch%3Amain)
+[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.7562790.svg)](https://doi.org/10.5281/zenodo.7562790)
```
Comment (fdumais) [inline README.md]: Can you add this Zenodo also https://zenodo.org/record/7562635#.Y-EXWNLMJH4

File: tractolearn/tractoio/dataset_fetch.py [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+    MNI2009CNONLINSYMM_ANAT = "mni2009cnonlinsymm_anat"
+    TRACTOINFERNO_HCP_REF_TRACTOGRAPHY = "tractoinferno_hcp_ref_tractography"
+    BUNDLE_LABEL_CONFIG = "bundle_label_config"
+    TRACTOINFERNO_HCP_CONTRASTIVE_THR_CONFIG = "tractoinferno_hcp_contrastive_thr_config"
+    GENERATIVE_STRML_MAX_COUNT_CONFIG = "generative_streamline_max_count_config"
+    GENERATIVE_STRML_RQ_COUNT_CONFIG = "generative_streamline_req_count_config"
+    GENERATIVE_SEED_STRML_RATIO_CONFIG = "generative_seed_streamline_ratio_config"
+    GENERATIVE_LOA_CONE_CONFIG = "generative_loa_cone_config"
+    GENERATIVE_WM_TISSUE_CRITERION_CONFIG = "generative_wm_tisue_criterion_config"
+
```
Comment (fdumais) [inline tractolearn/tractoio/dataset_fetch.py:39]: Can you fetch and unzip François atlas: `atlas.zip` -> https://zenodo.org/record/7562635#.Y-EXWNLMJH4

**Comment by fdumais** [comment]
Also, can you add the fetch_data script into the `pyproject.toml`

File: tractolearn/tractoio/dataset_fetch.py (hunk 5 above)
Comment (jhlegarreta) [reply tractolearn/tractoio/dataset_fetch.py:39]: Done in 063e1d3.

File: README.md (hunk 4 above)
Comment (jhlegarreta) [reply README.md]: Done in 063e1d3.

**Comment by jhlegarreta** [comment]
> Also, can you add the fetch_data script into the pyproject.toml

Done in 063e1d3.

**Review by fdumais — APPROVED** [review]

File: tractolearn/tractoio/dataset_fetch.py (hunk 3 above)
Comment (jhlegarreta) [reply tractolearn/tractoio/dataset_fetch.py]: Done in 80d4b8b.

**Pull request merged**